src/NuGet.Core/NuGet.Commands/RestoreCommand/Utility/MSBuildRestoreUtility.cs

```diff
@@ -945,36 +945,32 @@ public static RestoreAuditProperties GetRestoreAuditProperties(IMSBuildItem spec
 
         public static NuGetVersion GetSdkAnalysisLevel(string sdkAnalysisLevel)
         {
-            if (!string.IsNullOrEmpty(sdkAnalysisLevel))
+            if (string.IsNullOrEmpty(sdkAnalysisLevel))
             {
-                if (NuGetVersion.TryParse(sdkAnalysisLevel, out NuGetVersion version))
-                {
-                    return version;
-                }
-                else
-                {
-                    throw new ArgumentException(string.Format(CultureInfo.CurrentCulture, Strings.Invalid_AttributeValue, "SdkAnalysisLevel", sdkAnalysisLevel, "9.0.100"));
-                }
+                return null;
             }
 
-            return null;
+            if (!NuGetVersion.TryParse(sdkAnalysisLevel, out NuGetVersion version))
+            {
+                throw new ArgumentException(string.Format(CultureInfo.CurrentCulture, Strings.Invalid_AttributeValue, "SdkAnalysisLevel", sdkAnalysisLevel, "9.0.100"));
+            }
+
+            return version;
         }
 
         public static bool GetUsingMicrosoftNETSdk(string usingMicrosoftNETSdk)
         {
-            if (!string.IsNullOrEmpty(usingMicrosoftNETSdk))
+            if (string.IsNullOrEmpty(usingMicrosoftNETSdk))
```

Comment:
Given that nothing is default to false (correctly), I think the smae thing needs to be done to ProjectRestoreMetadata

Reply:
I hace removed the default true value. Which means the default is false.